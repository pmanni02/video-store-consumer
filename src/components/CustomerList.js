import React, { Component } from 'react';
import Customer from './Customer';
import PropTypes from 'prop-types';
import axios from 'axios';
import './CustomerList.css'

class CustomerList extends Component {
  constructor() {
    super();
    this.state = {
      customers: [],
    }
  }

  componentDidMount = () => {
    let query =  'http://localhost:3000/customers'
    console.log(query)

    axios.get(query)
      .then((response) => {
        console.log(response);
        this.setState({
          customers: response.data
        })
      })
      .catch((error) => {
        this.setState({
          error: error.message
        });
        if (error.message) {
          this.renderError();
          console.log(error.message);
        }
      }
    );
  }

  pickCustomer = (id, name) => {
    this.props.pickCustomerDetailCallback(id, name)
  }

  renderCustomerList = () => {
    const customerList = this.state.customers.map((item, index) => {
      return(
        <Customer
          key={index}
          id={item.id}
          name={item.name}
          address={item.address}
          city={item.city}
          state={item.state}
          postal_code={item.postal_code}
          phone={item.phone}
          customerCallback={this.pickCustomer}
        />
      )
    })
    return customerList
  }

  renderError = () => {
    return (
      <p className='validation-errors-dislay'>
        {this.state.error}
      </p>
    )
  }

  render() {
    return (
      <div className="customer-list">
        {this.renderCustomerList()}
      </div>
    )
  }
}

export default CustomerList;

CustomerList.propTypes = {
  pickCustomerDetailCallback: PropTypes.func,
}
