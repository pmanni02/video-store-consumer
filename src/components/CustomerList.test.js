import React from 'react';
import { mount } from 'enzyme';
import CustomerList from './CustomerList';

describe('CustomerList', () => {
  test('snapshot', () => {
    const wrapper = mount(
      <CustomerList
        pickCustomerDetailCallback={()=>{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
