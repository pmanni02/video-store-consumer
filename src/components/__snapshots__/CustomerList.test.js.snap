// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerList snapshot 1`] = `
<CustomerList
  pickCustomerDetailCallback={[Function]}
>
  <div
    className="customer-list"
  />
</CustomerList>
`;
